### PR TITLE
Added Install Font button when selecting a font file

### PR DESCRIPTION
**Pull request by puppetsw**
**Resolved / Related Issues**
Items resolved / related issues by this PR.
- Related #7800

**Details of Changes**
Add details of changes here.
- Added IsFontFile extension helper.
- Added resource string (Install).
- Corrected spelling of HasAdditionalActions. 
- Added messager handler to install font (could be used for installing infs at a later stage)

**Validation**
How did you test these changes?
- [x] Built and ran the app
- [ ] Tested the changes for accessibility

**Screenshots (optional)**
image

**Comments**
Toolbar text might be better as Install font. I just went with what Windows had originally in the context menu.

### CODE REVIEW

File: src/Files/UserControls/InnerNavigationToolbar.xaml [hunk 1]
```diff
@@ -285,6 +284,17 @@
...
                 </AppBarButton>
+                <AppBarButton
+                    x:Name="InstallFontButton"
+                    x:Load="{x:Bind ViewModel.IsFont, Mode=OneWay, FallbackValue=False}"
+                    Command="{x:Bind ViewModel.InstallFontCommand, Mode=OneWay}"
+                    Label="{helpers:ResourceString Name=Install}"
+                    LabelPosition="Default"
+                    ToolTipService.ToolTip="{helpers:ResourceString Name=Install}">
+                    <AppBarButton.Icon>
+                        <FontIcon Glyph="&#xE8D2;" />
```
Comment (yaichenbaum) [inline src/Files/UserControls/InnerNavigationToolbar.xaml:295]:
```suggestion
                        <FontIcon Glyph="&#xE185;;" />
```

File: src/Files.Launcher/MessageHandlers/InstallOperationsHandler.cs [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+                {
+                    var filePath = (string)message["filepath"];
+                    var fileExtension = (string)message["extension"];
+                    var isFont = new[] { ".fon", ".otf", ".ttc", ".ttf" }.Contains(fileExtension, StringComparer.OrdinalIgnoreCase);
+
+                    if (isFont)
+                    {
+                        var userFontDir = Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.LocalApplicationData), "Microsoft", "Windows", "Fonts");
+                        var destName = Path.Combine(userFontDir, Path.GetFileName(filePath));
+                        Win32API.RunPowershellCommand($"-command \"Copy-Item '{filePath}' '{userFontDir}'; New-ItemProperty -Name '{Path.GetFileNameWithoutExtension(filePath)}' -Path 'HKCU:\\Software\\Microsoft\\Windows NT\\CurrentVersion\\Fonts' -PropertyType string -Value '{destName}'\"", false);
```
Comment (yaichenbaum) [inline src/Files.Launcher/MessageHandlers/InstallOperationsHandler.cs:41]: @gave92 is there a way to use the shell api for this?

Comment (puppetsw) [reply src/Files.Launcher/MessageHandlers/InstallOperationsHandler.cs:41]: @yaichenbaum Did you mean using this for example? https://docs.microsoft.com/en-gb/windows/win32/api/wingdi/nf-wingdi-addfontresourcea?redirectedfrom=MSDN

**Comment by jiejasonliu** [comment]
The [documentation](https://docs.microsoft.com/en-us/windows/win32/gdi/wm-fontchange#remarks) says to send `WM_FONTCHANGE` to top-level processes for any font changes. 

It's probably a good idea to do that here if we stick with installing fonts without a dedicated API.

File: src/Files.Launcher/MessageHandlers/InstallOperationsHandler.cs (hunk 2 above)
Comment (gave92) [reply src/Files.Launcher/MessageHandlers/InstallOperationsHandler.cs:41]: Not that I'm aware of: using context menu does not work, using AddFontEesource does not persist font after reboot.

**Review by gave92 — APPROVED** [review]

**Comment by gave92** [comment]
Great work, merely changed the `IsFont` property to verify that _all_ selected items are font files. Also the commands must be defined in ColumnShellPage as well so they work in Column view.

**Comment by gave92** [comment]
In the future perhaps we should consider adding some feedback when using toolbar buttons, currently I'm tempted to click "Install font" repeatedly because nothing seem to happen xD

**Comment by yaichenbaum** [comment]
> In the future perhaps we should consider adding some feedback when using toolbar buttons, currently I'm tempted to click "Install font" repeatedly because nothing seem to happen =)

@gave92 I guess the button can be disabled and a progress ring can be show inside of it instead.

**Review by yaichenbaum — APPROVED** [review]

**Pull request merged**